mobile/src/navigations/root/root.tsx

```diff
@@ -22,41 +22,53 @@ const screenOptions: NativeStackNavigationOptions = {
 
 const Root: React.FC = () => {
   const dispatch = useAppDispatch();
-  const { isSurveyCompleted, authenticatedUser, authenticatedUserDataStatus } =
-    useAppSelector(({ auth }) => {
-      return {
-        isSurveyCompleted: auth.authenticatedUser?.isSurveyCompleted,
-        authenticatedUser: auth.authenticatedUser,
-        authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
-      };
-    });
+  const {
+    isSurveyCompleted,
+    authenticatedUser,
+    authenticatedUserDataStatus,
+    surveyPreferencesDataStatus,
+  } = useAppSelector(({ auth }) => {
+    return {
+      isSurveyCompleted: auth.authenticatedUser?.isSurveyCompleted,
+      authenticatedUser: auth.authenticatedUser,
+      authenticatedUserDataStatus: auth.authenticatedUserDataStatus,
+      surveyPreferencesDataStatus: auth.surveyPreferencesDataStatus,
+    };
+  });
+
+  const isLoaderVisible =
+    authenticatedUserDataStatus === DataStatus.IDLE ||
+    authenticatedUserDataStatus === DataStatus.PENDING ||
+    surveyPreferencesDataStatus === DataStatus.PENDING;
 
   useEffect(() => {
     void dispatch(authActions.getAuthenticatedUser());
   }, [dispatch]);
 
-  if (
-    authenticatedUserDataStatus === DataStatus.IDLE ||
-    authenticatedUserDataStatus === DataStatus.PENDING
-  ) {
-    return <Loader />;
-  }
-
-  return (
-    <NativeStack.Navigator screenOptions={screenOptions}>
-      {authenticatedUser ? (
-        isSurveyCompleted ? (
-          <NativeStack.Screen name={RootScreenName.MAIN} component={Main} />
-        ) : (
-          <NativeStack.Screen name={RootScreenName.SURVEY} component={Survey} />
-        )
+  const getScreen = (): JSX.Element => {
+    if (authenticatedUser) {
```

Comment:
Please decrease complexity with it, create a config(constant) with all screens, the object contains name, component and conditionToRender or simething like that. And Helper should receive it and condtitions like authenticatedUser and isSurveyCompleted

Reply:
I tried to do it this way:
```
const screensConfig = [
  {
    name: RootScreenName.MAIN,
    component: Main,
    conditionToRender: (
      authenticatedUser: boolean,
      isSurveyCompleted: boolean,
    ) => authenticatedUser && isSurveyCompleted,
  },
  {
    name: RootScreenName.SURVEY,
    component: Survey,
    conditionToRender: (
      authenticatedUser: boolean,
      isSurveyCompleted: boolean,
    ) => authenticatedUser && !isSurveyCompleted,
  },
  {
    name: RootScreenName.SIGN_IN,
    component: Auth,
    conditionToRender: (authenticatedUser: boolean) => !authenticatedUser,
  },
  {
    name: RootScreenName.SIGN_UP,
    component: Auth,
    conditionToRender: (authenticatedUser: boolean) => !authenticatedUser,
  },
];
```
```
  const getScreenToRender = (
    authenticatedUser: boolean,
    isSurveyCompleted: boolean,
  ) => {
    for (const screen of screensConfig) {
      if (screen.conditionToRender(authenticatedUser, isSurveyCompleted)) {
        return (
          <NativeStack.Screen
            name={screen.name}
            component={screen.component}
            key={screen.name}
          />
        );
      }
    }
  };
```
, and everything works everywhere except the Sign Up page. I'm getting this error: 

_"The action 'NAVIGATE with payload ("name*:"Sign
Up'"'path"."/Sign Up'" params":O) was not handled by any navigator.
Do you have a screen named 'Sign Up'?"_

I've been trying to solve it for a while, but nothing helps. Any ideas?

Reply:
I suppose return in for of ends the iteration at first truthy condition. Let's replace with .filter

Reply:
Thank you)